pkg/templates/reader.go

```diff
@@ -17,39 +17,33 @@ package templates
 import (
 	"fmt"
 
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1"
-	"github.com/vmware-tanzu/cartographer/pkg/eval"
 )
 
-const PREVIEW_CHARACTER_LIMIT = 1024
+type Reader interface {
```

Comment:
It's quite a "go"-ism, but, usually single-method interfaces are named after the method + `er`, e.g., [`io.Reader`](https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/io/io.go;l=83) provides the `Read` method

```go
type Reader interface {
	Read(p [][byte](https://pkg.go.dev/builtin#byte)) (n [int](https://pkg.go.dev/builtin#int), err [error](https://pkg.go.dev/builtin#error))
}
```

and _usually_ rather than getters being named "GetFoo", they're _usually_ plain "`Foo()`" which in this case could look like

```go
// uf, `outputter` looks like a rough name? not sure exactly 
//
type Outputter interface {
        Output(stampedObject *unstructured.Unstructured) (*templates.Output, error)
}
```

wdyt?

Reply:
ps.: i'm personally not that strongly against explicit `Get<Foo>()` methods in interfaces

Reply:
im fine with outputter